src/main/java/service/impl/DataConverterImpl.java

```
@@ -0,0 +1,21 @@
+package service.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+import model.FruitTransaction;
+import service.DataConverter;
+
+public class DataConverterImpl implements DataConverter {
+    @Override
+    public List<FruitTransaction> convertToTransactions(List<String> lines) {
```

Method `convertToTransactions` does not validate its `lines` parameter. The task requires handling null parameters (e.g. "3. Null parameters") and failing fast with informative messages; add a null check and throw an informative RuntimeException if `lines` is null. See task guidance about validating inputs .